### PR TITLE
Fixing slot integration, improving pickup workflow feedback, and adding global error display.

### PR DESCRIPTION
**Pull request by merivale**
This was going to just sort out the currentSlot behaviour (following #3004). But while I was at it, I thought I should add some global error feedback to the user when the slots go out of sync. And before I had committed, I found myself improving the feedback of the wallet pickup workflow - which is unrelated to the rest of the PR, but needs doing.

### CODE REVIEW

File: marlowe-dashboard-client/src/MainFrame/State.purs [hunk 1]
```diff
@@ -261,8 +268,9 @@ handleAction (PickupAction pickupAction) = case pickupAction of
           handleAction AddNewWallet
           handleAction $ PickupAction $ Pickup.PickupWallet walletDetails
   Pickup.PickupWallet walletDetails -> do
-    -- TODO: set some state flag to "picking up..." so that we can notify the user in the view
-    -- that something is happening (because all these requests may take a while)
+    -- Picking up a wallet takes a while (several trips to the PAB are required). So first of all we set
+    -- the pickingUp flag to true, so that we can inform the user that this is all ongoing...
+    assign (_pickupState <<< _pickingUp) true
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/MainFrame/State.purs:273]: In a previous PR I saw you changed from RemoteData to an Either, but now we are introducing a boolean to indicate loading... I'm wondering if we should use RemoteData back again.

File: marlowe-dashboard-client/src/Pickup/View.purs [hunk 2]
```diff
@@ -132,15 +129,15 @@ pickupNewWalletCard wallets newWalletDetails pickupNewWalletContext =
               [ text "Cancel" ]
           , button
               [ classNames $ Css.primaryButton <> [ "flex-1" ]
-              , disabled $ isJust mWalletNicknameError || isJust mContractInstanceIdError
+              , disabled $ isJust mWalletNicknameError || isJust mContractInstanceIdError || pickingUp
               , onClick_ PickupNewWallet
               ]
-              [ text "Pickup" ]
+              [ text if pickingUp then "Picking up... " else "Pickup" ]
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Pickup/View.purs:135]: Besides the informative text, there are some designs Russell provided for loading stuff, which are a perfect match for this already available tailwind animation https://tailwindcss.com/docs/animation#pulse

File: marlowe-dashboard-client/src/Play/State.purs [hunk 3]
```diff
@@ -94,12 +100,15 @@ handleAction CloseCard = do
...
-  modify_
-    $ set _currentSlot currentSlot
+handleAction (SetCurrentSlot slot) = do
+  toContractHome $ ContractHome.handleAction $ ContractHome.AdvanceTimedOutContracts slot
+  assign _currentSlot slot
+
+handleAction (CheckCurrentSlot slot) = do
+  currentSlot <- use _currentSlot
+  let
+    maxToleratedDifference = Slot $ fromInt 2
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Play/State.purs:110]: I think up to 10 seconds we should be ok... 2 will probably cause a lot of false positives

File: marlowe-dashboard-client/src/Play/View.purs [hunk 4]
```diff
@@ -169,22 +170,42 @@ renderCard wallets newWalletDetails templates playState card =
...
     contractsState = view _contractsState playState
   in
     div
       -- TODO: Revisit the overflow-auto... I think the scrolling should be set at the screen level and not here.
       --       this should have overflow-hidden to avoid the main div to occupy more space than available.
-      [ classNames [ "absolute", "top-0", "bottom-0", "left-0", "right-0", "overflow-auto", "z-0" ] ] case screen of
-      ContractsScreen -> [ ContractHomeAction <$> contractsScreen currentSlot contractsState ]
-      WalletLibraryScreen -> [ walletLibraryScreen wallets ]
-      TemplateScreen -> [ TemplateAction <$> contractSetupScreen wallets currentSlot templateState ]
+      [ classNames [ "absolute", "top-0", "bottom-0", "left-0", "right-0", "overflow-auto", "z-0" ] ]
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Play/View.purs:187]: setting all top, bottom, properties is the same as using `inset` 

https://tailwindcss.com/docs/top-right-bottom-left

**Review by hrajchert — APPROVED** [review]
LGTM

File: marlowe-dashboard-client/src/MainFrame/State.purs (hunk 1 above)
Comment (merivale) [reply marlowe-dashboard-client/src/MainFrame/State.purs:251]: I was wondering that too, but it wasn't obvious to me which property should be RemoteData in this case. I'll give it some more thought for the next PR.

**Pull request merged**